CommunityPromotionScreen/Src/NewPromotionScreenbyDefault/Classes/NPSBDP_UIArmory_PromotionHero.uc

```diff
@@ -567,35 +569,65 @@ function bool CanPurchaseAbilityEx(int Rank, int Branch, name AbilityName, out s
 		// Don't allow non hero units to purchase abilities with AP without a training center
 		// The branch is checked here so that "No Training Center" takes priority over "No
 		// class perk picked" as a reason.
-		if((UnitState.HasPurchasedPerkAtRank(Rank) || Branch >= AbilityRanks) && !UnitState.IsResistanceHero() && !CanSpendAP())
+		if ((UnitState.HasPurchasedPerkAtRank(Rank) || Branch >= AbilityRanks) && !UnitState.IsResistanceHero() && !CanSpendAP())
 		{
-			ReasonLockedText = ReasonNoTrainingCenter;
-			return false;
-		}
-
-		// Don't allow non hero units to purchase abilities on the xcom perk row before getting a rankup perk
-		if(!UnitState.HasPurchasedPerkAtRank(Rank) && !UnitState.IsResistanceHero() && Branch >= AbilityRanks )
+			strLocReasonLocked = ReasonNoTrainingCenter;
+			nReasonLocked = 'NoTrainingCenter';
+		} // Don't allow non hero units to purchase abilities on the xcom perk row before getting a rankup perk
+		else if (!UnitState.HasPurchasedPerkAtRank(Rank) && !UnitState.IsResistanceHero() && Branch >= AbilityRanks )
 		{
-			ReasonLockedText = ReasonNoClassPerkPurchased;
-			return false;
+			strLocReasonLocked = ReasonNoClassPerkPurchased;
+			nReasonLocked = 'NoClassPerkPurchased';
 		}
-	}
-
-	// Normal behaviour and hero promotion emulation (ability ranks == 0)
-	if (Rank >= UnitState.GetRank())
+	} // Normal behaviour and hero promotion emulation (ability ranks == 0)
+	else if (Rank >= UnitState.GetRank())
 	{
-		ReasonLockedText = ReasonNotHighEnoughRank;
+		strLocReasonLocked = ReasonNotHighEnoughRank;
+		nReasonLocked = 'NotHighEnoughRank';
 	}
 	else if (!CanAffordAbility(Rank, Branch))
 	{
-		ReasonLockedText = ReasonNotEnoughAP;
+		strLocReasonLocked = ReasonNotEnoughAP;
+		nReasonLocked = 'NotEnoughAP';
 	}
 	else if (!UnitState.MeetsAbilityPrerequisites(AbilityName))
 	{
-		ReasonLockedText = ReasonLacksPrerequisites;
+		strLocReasonLocked = ReasonLacksPrerequisites;
+		nReasonLocked = 'LacksPrerequisites';
 	}
 
-	return ReasonLockedText == "";
+	// Start Issue #3
+	/// Mods can listen to the 'OverrideCanPurchaseAbility' event to use their own logic 
+	/// to determine whether this particular unit should be able to unlock this particular ability.
+	/// Tuple passes:
+	/// strLocReasonLocked - localized string that will be displayed as a reason why this ability cannot be unlocked.
+	/// nReasonLocked is passed in the Tuple so mods can see *why* NPSBD's considers this ability cannot be unlocked
+	/// without having to check the strLocReasonLocked localized string.
+	/// nReasonLocked can take the following values: 
+	/// 'NoTrainingCenter', 'NoClassPerkPurchased', 'NotHighEnoughRank', 'NotEnoughAP', 'LacksPrerequisites'.
```

Comment:
If we're going for CHL doc syntax, then this would be a bit easier to read as:

```suggestion
	///  - 'NoTrainingCenter'
	///  - 'NoClassPerkPurchased'
	///  - 'NotHighEnoughRank'
	///  - 'NotEnoughAP'
	///  - 'LacksPrerequisites'
```

I would also consider using backticks for all references to variable names in the rest of the doc comment, just so that they stand out as code. Minor preference, though.